src/data_structures/setup.rs

```diff
@@ -355,8 +356,7 @@ impl GenericSetupStorage<CosetEvaluations> {
 
 pub struct SetupCache {
     monomials: GenericSetupStorage<MonomialBasis>,
-    cosets: Vec<Option<GenericSetupStorage<CosetEvaluations>>>,
-    tmp_coset: GenericSetupStorage<CosetEvaluations>,
+    cosets: Vec<Option<Rc<GenericSetupStorage<CosetEvaluations>>>>,
```

Comment:
also (i realize this is a preexisting condition) why are there lifetime annotations on ArgumentCache and QuotientCache above but no lifetime annotations for SetupCache and TraceCache?

Reply:
I guess that is a question for @saitima.

Reply:
I don't consider this to be in scope of this PR so i am marjing this as resolved.